lib/node/NodeEnvironmentPlugin.js

```diff
@@ -18,12 +19,31 @@ class NodeEnvironmentPlugin {
 	 * @returns {void}
 	 */
 	apply(compiler) {
+		const isMemoryFS = compiler.options.memory ? true : false;	
+		const {MemoryFileSystem} = isMemoryFS ? new NodeMemoryFileSystem() : null;
+		if (isMemoryFS) {
+			// @ts-ignore
+			try {
+				const fs = require('fs');
+				fs.statSync(compiler.options.entry);
+			} catch(err) {
+				if(err.code === 'ENOENT') {
```

Comment:
Is any other (unknown) error just ignored?

Reply:
most memory errors as up now are gone.
the best solution is an isolated environment and a dynamic working tree with 2 or eve 3 .ignore files
that pretty much solv 90% of the problem.
the rest of the debugging would have to start in stop using sync proc